semgrep/semgrep/evaluation.py

```diff
@@ -137,39 +137,35 @@ def _evaluate_single_expression(
         raise UnknownOperatorError(f"unknown operator {expression.operator}")
 
 
-# Given a `where-python` expression as a string and currently matched metavars,
-# return whether the expression matches as a boolean
 def _where_python_statement_matches(
     where_expression: str, metavars: Dict[str, Any]
 ) -> bool:
     # TODO: filter out obvious dangerous things here
-    output_var = None
+    result = False
 
-    # HACK: we're executing arbitrary Python in the where-python,
-    # be careful my friend
-    vars = {k: v["abstract_content"] for k, v in metavars.items()}
+    local_vars = {k: v["abstract_content"] for k, v in metavars.items()}
     RETURN_VAR = "semgrep_pattern_return"
     try:
         cleaned_where_expression = where_expression.strip()
         lines = cleaned_where_expression.split("\n")
         new_last_line = f"{RETURN_VAR} = {lines[-1]}"
         lines[-1] = new_last_line
         to_eval = "\n".join(lines)
-        scope = {"vars": vars}
+        scope = {"vars": local_vars}
         # fmt: off
         exec(to_eval, scope)  # nosem: contrib.dlint.dlint-equivalent.insecure-exec-use, python.lang.security.audit.exec-detected.exec-detected
         # fmt: on
-        output_var = scope[RETURN_VAR]
+        result = scope[RETURN_VAR]  # type: ignore
```

Comment:
It's impossible to statically determine the type here, the `isinstance` call below confirms the type for us :+1: